tests/test_vcs_hg/test_vcs_hg.py

```diff
@@ -0,0 +1,68 @@
+from os.path import dirname, join
+import subprocess
+
+from dxr.testing import DxrInstanceTestCase
+
+from nose import SkipTest
+from nose.tools import ok_
+
+
+class MercurialTests(DxrInstanceTestCase):
+    """Test our Mercurial integration, both core and omniglot."""
+
+    @classmethod
+    def setup_class(cls):
+        """hg changes its state files during the tests. We extract from an
+        archive so that git doesn't flag them as changes.
+
+        """
+        build_dir = join(dirname(__file__), 'code')
+        subprocess.check_call(['make'], cwd=build_dir)
+        super(cls, MercurialTests).setup_class()
+
+    @classmethod
+    def teardown_class(cls):
+        """We delete the .hg directory we extracted before.
+
+        """
+        build_dir = join(dirname(__file__), 'code')
+        subprocess.check_call(['make', 'clean'], cwd=build_dir)
```

Comment:
If the aforementioned uses of build_command clean_command work out, then this can go away too. In any case, it should be factored up, though.